### PR TITLE
[3.4] Refactoring_tests: alarm, customer

### PR DESCRIPTION
**Pull request by nickAS21**
## Pull Request description

Tests:
- alarm  

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.
 
## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

File: application/src/test/java/org/thingsboard/server/controller/AbstractNotifyEntityTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+    protected void testNotifyEntityOne(HasName entity, EntityId entityId, EntityId originatorId,
+                                       TenantId tenantId, CustomerId customerId, UserId userId, String userName,
+                                       ActionType actionType, Object... additionalInfo) {
+        testSendNotificationMsgToEdgeServiceOne(entityId, tenantId, actionType);
+        testLogEntityActionOne(entity, originatorId, tenantId, customerId, userId, userName, actionType, additionalInfo);
+        testPushMsgToRuleEngineOne(originatorId, tenantId);
+        Mockito.reset(tbClusterService, auditLogService);
+    }
+
```
Comment (YevhenBondarenko) [inline application/src/test/java/org/thingsboard/server/controller/AbstractNotifyEntityTest.java:58]: remove extra line

File: application/src/test/java/org/thingsboard/server/controller/AbstractNotifyEntityTest.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        T result = null;
+        if (additionalInfo != null && additionalInfo.length > index) {
+            Object paramObject = additionalInfo[index];
+            if (clazz.isInstance(paramObject)) {
+                result = clazz.cast(paramObject);
+            }
+        }
+        return result;
+    }
+
```
Comment (YevhenBondarenko) [inline application/src/test/java/org/thingsboard/server/controller/AbstractNotifyEntityTest.java:175]: remove extra line

File: application/src/test/java/org/thingsboard/server/controller/BaseCustomerControllerTest.java [hunk 3]
```diff
@@ -159,36 +255,19 @@ public void testDeleteCustomer() throws Exception {
...
+        testNotifyEntityBroadcastEntityStateChangeEventOneMsgToEdgeServiceNever(savedCustomer, savedCustomer.getId(), savedCustomer.getId(),
+                savedCustomer.getTenantId(), savedCustomer.getId(), tenantAdmin.getId(), tenantAdmin.getEmail(),
+                ActionType.DELETED, savedCustomer.getId().getId().toString());
+
         doGet("/api/customer/" + savedCustomer.getId().getId().toString())
                 .andExpect(status().isNotFound());
     }
 
-    @Test
-    public void testSaveCustomerWithEmptyTitle() throws Exception {
```
Comment (YevhenBondarenko) [inline application/src/test/java/org/thingsboard/server/controller/BaseCustomerControllerTest.java:170]: why you removed this tests?

File: application/src/test/java/org/thingsboard/server/controller/AbstractNotifyEntityTest.java (hunk 1 above)
Comment (nickAS21) [reply application/src/test/java/org/thingsboard/server/controller/AbstractNotifyEntityTest.java:58]: Ok

File: application/src/test/java/org/thingsboard/server/controller/AbstractNotifyEntityTest.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                    Mockito.any(exp.getClass()), Mockito.eq(additionalInfo));
+        } else {
+            Mockito.verify(auditLogService, times(1)).logEntityAction(Mockito.eq(tenantId),
+                    Mockito.eq(customer_NULL_UUID), Mockito.eq(userId), Mockito.eq(userName),
+                    Mockito.eq(entity_NULL_UUID), Mockito.any(entity.getClass()), Mockito.eq(actionType),
+                    Mockito.any(exp.getClass()));
+        }
+        Mockito.verify(tbClusterService, never()).pushMsgToRuleEngine(Mockito.any(), Mockito.any(entity_NULL_UUID.getClass()),
+                Mockito.any(), Mockito.any());
+        Mockito.reset(tbClusterService, auditLogService);
```
[the remark this answers is not in the document]
Comment (nickAS21) [reply application/src/test/java/org/thingsboard/server/controller/AbstractNotifyEntityTest.java:110]: Mockito.reset(tbClusterService, auditLogService);

File: application/src/test/java/org/thingsboard/server/controller/AbstractNotifyEntityTest.java (hunk 2 above)
Comment (nickAS21) [reply application/src/test/java/org/thingsboard/server/controller/AbstractNotifyEntityTest.java:175]: OK

File: application/src/test/java/org/thingsboard/server/controller/BaseAlarmControllerTest.java [hunk 5]
```diff
@@ -56,159 +60,343 @@ public void teardown() throws Exception {
...
 
     @Test
     public void testDeleteAlarmViaDifferentTenant() throws Exception {
         loginTenantAdmin();
+
+        Mockito.reset(tbClusterService, auditLogService);
+
         Alarm alarm = createAlarm(TEST_ALARM_TYPE);
+
+        testNotifyEntityOne(alarm, alarm.getId(), alarm.getOriginator(),
```
[the remark this answers is not in the document]
Comment (nickAS21) [reply application/src/test/java/org/thingsboard/server/controller/BaseAlarmControllerTest.java:216]: In each test, as a rule, an alarm is created with different Actions and then either UPDATE or another tenant
Tests highlight these facts.

File: application/src/test/java/org/thingsboard/server/controller/BaseCustomerControllerTest.java (hunk 3 above)
Comment (nickAS21) [reply application/src/test/java/org/thingsboard/server/controller/BaseCustomerControllerTest.java:170]: remove to testSaveCustomerWithViolationOfValidation()

File: application/src/test/java/org/thingsboard/server/controller/BaseAlarmControllerTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (nickAS21) [reply application/src/test/java/org/thingsboard/server/controller/BaseAlarmControllerTest.java]: Ок

File: application/src/test/java/org/thingsboard/server/controller/BaseCustomerControllerTest.java (hunk 3 above)
Comment (nickAS21) [reply application/src/test/java/org/thingsboard/server/controller/BaseCustomerControllerTest.java:170]: Ок

**Review by YevhenBondarenko — APPROVED** [review]

**Pull request merged**